generators/entity-client/templates/angular/src/main/webapp/app/entities/_entity-management-dialog.component.ts

```diff
@@ -61,26 +61,32 @@ import { <%= uniqueRel.otherEntityAngularName%>Service } from '../<%= uniqueRel.
         }
     }
 }); _%>
+<%_ if ( fieldsContainInstant || fieldsContainZonedDateTime ) { _%>
+import { StringToMomentPipe } from '../../shared/util/string-to-moment-pipe';
+import { MomentToStringPipe } from '../../shared/util/moment-to-string-pipe';
+<%_ } _%>
 
 @Component({
     selector: '<%= jhiPrefixDashed %>-<%= entityFileName %>-dialog',
     templateUrl: './<%= entityFileName %>-dialog.component.html'
 })
 export class <%= entityAngularName %>DialogComponent implements OnInit {
 
-    <%= entityInstance %>: I<%= entityAngularName %>;
+    private _<%= entityInstance %>: I<%= entityAngularName %>;
     isSaving: boolean;
     <%_
-    for (const idx in variables) { %>
+    for ( const idx in variables ) { %>
     <%- variables[idx] %>
     <%_ } _%>
-    <%_ for (idx in fields) {
+    <%_ for ( idx in fields ) {
         const fieldName = fields[idx].fieldName;
         const fieldType = fields[idx].fieldType;
-        if (fieldType === 'LocalDate') { _%>
-    <%= fieldName %>Dp: any;
-        <%_ }
-    } _%>
+        if ( fieldType === 'LocalDate' ) { _%>
+    <%= fieldName %>Dp: any; %>
+    <%_ } else if ( ['Instant', 'ZonedDateTime'].includes(fieldType) ) { _%>
+    <%= fieldName: string; %>
```

Comment:
Your EJS is a little messed up, `<%= fieldName: string; %>` -> `<%= fieldName %>: string;`